logging/google/cloud/logging/handlers/GAEHandler.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Logging handler for App Engine Flexible
+
+Logs to the well-known file that the fluentd sidecar container on App Engine
+Flexible is configured to read from and send to Stackdriver Logging.
+
+This file is largely copied from:
+
+https://github.com/GoogleCloudPlatform/python-compat-runtime/blob/master/appengine-vmruntime/vmruntime/cloud_logging.py
+
+"""
+
+import json
+import logging.handlers
+import math
+import os
+
+LOG_PATH_TEMPLATE = '/var/log/app_engine/app.{pid}.json'
+MAX_LOG_BYTES = 128 * 1024 * 1024
+LOG_FILE_COUNT = 3
+
+
+class GAEHandler(logging.handlers.RotatingFileHandler):
+    """A handler that writes to the GAE fluentd Stackdriver logging file.
+
+    Writes to the file that the fluentd agent on App Engine Flexbile is
+    configured to discover logs and send them to  Stackdriver Logging.
+    Log entries are wrapped in JSON and with appropriate metadata. The
+    process of converting the user's formatted logs into a JSON payload for
+    Stackdriver Logging consumption is implemented as part of the handler
+    itself, and not as a formatting step, so as not to interfere with
+    user-defined logging formats.
+
+    The handler will look for HTTP header information in the environment (which
+    will be present in the GAE 1.0-compatible runtime) and, if it exists, will
+    parse the X-Cloud-Trace-Context header to add a Trace ID to each log
+    record.
+    Logging calls can also alternatively 'trace_id' in as a field in the
+    'extra' dict, which will be used preferentially to fill out the Trace ID
+    metadata.
+    """
+
+    def __init__(self):
+        # Large log entries will get mangled if multiple workers write to the
+        # same file simultaneously, so we'll use the worker's PID to pick a log
+        # filename.
```

Comment:
This is a pretty useful comment. 

1. Why not put it in the class docstring?
1. Is the `filename` accesible as a property on the handler instance?